src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -102,27 +102,25 @@ private function registerHandlers(ContainerBuilder $container)
         $handlerResolver->replaceArgument(0, ServiceLocatorTagPass::register($container, $handlersLocatorMapping));
     }
 
-    private function guessHandledClass(ContainerBuilder $container, string $serviceId): string
+    private function guessHandledClass(\ReflectionClass $handlerClass, string $serviceId): string
     {
-        $reflection = $container->getReflectionClass($container->getDefinition($serviceId)->getClass());
-
         try {
-            $method = $reflection->getMethod('__invoke');
+            $method = $handlerClass->getMethod('__invoke');
         } catch (\ReflectionException $e) {
-            throw new RuntimeException(sprintf('Service "%s" should have an `__invoke` function.', $serviceId));
+            throw new RuntimeException(sprintf('Invalid service "%s": class "%s" must have an "__invoke()" method.', $serviceId, $handlerClass->getName()));
```

Comment:
What do you think about replacing all the "Invalid service" by "Invalid handler service" to clarify why it requires such method?

Reply:
👍 done